Modules/_ctypes/cfield.c

```diff
@@ -252,17 +254,20 @@ static PyGetSetDef PyCField_getset[] = {
 };
 
 static int
-PyCField_traverse(CFieldObject *self, visitproc visit, void *arg)
+PyCField_traverse(PyObject *op, visitproc visit, void *arg)
 {
+    CFieldObject *self = _CFieldObject_CAST(op);
     Py_VISIT(Py_TYPE(self));
     Py_VISIT(self->proto);
     return 0;
 }
 
 static int
-PyCField_clear(CFieldObject *self)
+PyCField_clear(PyObject *op)
 {
+    CFieldObject *self = _CFieldObject_CAST(op);
     Py_CLEAR(self->proto);
+    Py_CLEAR(self->name);
```

Comment:
@encukou This one is not strictly needed per se ('name' is an exact PyUnicode according to the docs so we shouldn't have a cycle) but it allows me to simplify a bit `PyCField_dealloc` (namely, no need for a cast). Do you prefer that I put it back though?